src/lib/utils/thorchain/memo.test.ts

```diff
@@ -0,0 +1,685 @@
+import { describe, expect, it } from 'vitest'
+
+import { assertAndProcessMemo } from './memo' // Update this import path if necessary
+
+describe('assertAndProcessMemo', () => {
```

Comment:
Some of these tests feel a little redundant, like all of the `should gracefully handle being called multiple times in a row`, but this is a high risk helper so over-testing is an ok tradeoff here.

Reply:
Yeah we could probably consolidate and have multiple assertions in one, though I like the declarativeness of the many many tests personally!